django/db/models/expressions.py

```diff
@@ -1201,7 +1201,9 @@ def get_refs(self):
         return {self.refs}
 
     def relabeled_clone(self, relabels):
-        return self
+        clone = self.copy()
+        clone.source = self.source.relabeled_clone(relabels)
+        return clone
```

Comment:
Is this different from what the parent class does? If no, we could just remove this re-definition as a fix? (I removed this method and the newly added test keeps passing).

(I renamed `change_map` from the original code as`relabels` for clarity sake)

```python
    def relabeled_clone(self, relabels):
        clone = self.copy()
        clone.set_source_expressions(
            [
                e.relabeled_clone(relabels) if e is not None else None
                for e in self.get_source_expressions()
            ]
        )
        return clone
```

Reply:
`Ref` has only one `source`, I think it's worth keeping a custom implementation.